src/components/ModalOnEvent.tsx

```diff
@@ -55,7 +56,9 @@ const ModalOnEvent: React.FC<ModalOnEventProps> = ({
 
     return (
         <Modal cross={false} view={view} visible={modalVisible} onShow={onShow} onClose={onModalClose}>
-            {children}
+            {nullable(children, (c) =>
+                cloneElement(c as React.ReactElement, { ...(c as React.ReactElement).props, ...modalProps }),
```

Comment:
@awinogradov oh look here comes cloneElement )

Reply:
i don't know another way to pass props outside of React

Reply:
`const modalContext = createContext()`...

Reply:
cloneElement is a little bit simple instead of context 😄